src/contracts/Faucet/Faucet.ts

```diff
@@ -67,4 +71,66 @@ export class Faucet {
     await this.client.sendRawMessage(bytesToHex(encodedMessage));
     return message.hash();
   }
+
+  /**
+   * Withdraws the specified value to the given address with retries.
+   *
+   * @async
+   * @param {Hex} address The address to which the withdrawal should be made.
+   * @param {bigint} [value=1000000000000000000n] The value that should be withdrawn to the given address.
+   * @param {?number} [retry=3] How many times to retry the withdrawal in case of failure.
+   * @returns {Uint8Array} The hash of the withdrawal message.
+   */
+  async withdrawToWithRetry(
+    address: Hex,
+    value = 1000000000000000000n,
+    retry = 5,
+  ) {
+    let currentRetry = 0;
+    while (currentRetry++ < retry) {
+      try {
+        const [refinedSeqno, chainId] = await Promise.all([
+          this.client.getMessageCount(Faucet.address, "latest"),
+          this.client.chainId(),
+        ]);
+        const calldata = encodeFunctionData({
+          abi: FaucetAbi,
+          functionName: "withdrawTo",
+          args: [address.toLowerCase(), value],
+        });
+        const message = new ExternalMessageEnvelope({
+          isDeploy: false,
+          to: hexToBytes(Faucet.address),
+          chainId,
+          seqno: refinedSeqno,
+          data: hexToBytes(calldata),
+          authData: new Uint8Array(0),
+        });
+        const encodedMessage = message.encode();
+        await this.client.sendRawMessage(bytesToHex(encodedMessage));
+        const hash = bytesToHex(message.hash());
+        const receipts: IReceipt[] = await Promise.race([
+          new Promise<[]>((resolve) => setTimeout(() => resolve([]), 10000)),
```

Comment:
I suggest to use timeout property from client here

Reply:
Actually i see that it is the property of the transport, which is kinda low level of abstraction. We can not inherit this property from transport, it should work in the opposite direction. So maybe just pass timeout as parameter of `withdrawToWithRetry`?

Reply:
it's not about case when client is idling. It's about case when transaction is lost

Reply:
We could send transaction. But because a lot of checks happen not in a moment when rpc receiving message but in a moment later transaction could be just lost

Reply:
Yes, i got it. Can we add timeout parameter to withdrawToWithRetry?

Reply:
we can but right now don't want to have too much options here